psst-gui/src/ui/mod.rs

```diff
@@ -42,19 +43,31 @@ pub mod user;
 pub mod utils;
 
 pub fn main_window(config: &Config) -> WindowDesc<AppState> {
-    let mut win = WindowDesc::new(root_widget())
+    let mut win = WindowDesc::new(root_widget(config))
         .title(compute_main_window_title)
-        .window_size(config.window_size)
         .show_title(false)
         .transparent_titlebar(true);
+
     if config.kiosk_mode {
-        win = win.set_window_state(WindowState::Maximized)
-        .resizable(false)
-        .show_titlebar(false);
+        win = win
+            .set_window_state(WindowState::Maximized)
+            .resizable(false)
+            .show_titlebar(false)
+            .set_always_on_top(true);
+
+        // Set the window size to the primary monitor's work area and position it at (0, 0)
+        if let Some(monitor) = druid::Screen::get_monitors().first() {
+            let work_area = monitor.virtual_work_rect();
+            win = win
+                .window_size(work_area.size())
+                .set_position(druid::Point::new(0.0, 0.0));
+        }
     } else {
-        win = win.with_min_size((theme::grid(65.0), theme::grid(50.0)))
-
+        win = win
+            .window_size(config.window_size)
+            .with_min_size((theme::grid(65.0), theme::grid(50.0)));
     }
```

Comment:
@SO9010 In the kiosk mode, I'm setting the window state to maximize. Setting it such that it's always on top and then also setting the window size to the maximum window size of the monitor and putting the position at 0, 0.

The max window size was necessary for macOS, but it might not be necessary for Windows or Linux, I want to confirm with you that this still works on Windows/Linux. I also moved to set the default window size only when we aren't in kiosk mode.

Reply:
Thank you for this! We don't really want the main window to be always on top as when the preferences window that will not have priority over the window so it might or might not show! 

Also I was wandering, on Linux the login and preferences section is still small, this is due to your fix only being for the main route widget I believe.

I was thinking that due to it being kiosk mode the whole set up should also be in full screen!